tests/unit/viahtml/stats_test.py

```diff
@@ -0,0 +1,147 @@
+import json
+
+import pytest
+from h_matchers import Any
+from pkg_resources import resource_filename
+from requests import RequestException
+
+from viahtml.stats import UWSGINewRelicStatsGenerator
+
+STATS_ENDPOINT = "http://example.com"
+
+
+class TestUWSGINewRelicStatsGenerator:
+    def test_it_calls_the_end_point_and_returns_stats(self, stats, requests):
+        results = stats()
+
+        assert results == Any.generator.containing([("Custom/Alive/Metrics", 1)])
+        requests.get.assert_called_once_with(url=STATS_ENDPOINT, timeout=1)
```

Comment:
Interesting quirk: you have to make the `get` assert after comparison the results, otherwise the `generator` hasn't actually taken effect yet.

Calling the matcher will exhaust the generator.